### PR TITLE
feat(EMS-677): Company Details - Check your answers

### PR DESCRIPTION
**Pull request by Zainzzkk**
# Changes made:
* Added check your answers for company details section
* Added unit tests
* Added e2e test

### CODE REVIEW

File: e2e-tests/cypress/support/insurance/check-your-business-summary-list.js [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+  },
+  [PHONE_NUMBER]: () => {
+    const fieldId = PHONE_NUMBER;
+    const expectedKey = FIELDS[fieldId].SUMMARY.TITLE;
+
+    const expectedValue = application.EXPORTER_COMPANY[fieldId];
+
+    const expectedChangeLinkText = FIELDS[fieldId].SUMMARY.TITLE;
+
+    assertRow(fieldId, expectedKey, expectedValue, expectedChangeLinkText);
```
Comment (ttbarnes) [inline e2e-tests/cypress/support/insurance/check-your-business-summary-list.js:177]: Do you think we could make a helper function for each field e.g `getField`, passing the ID - which then returns they key, value and expected change link text - then pass thesee to `assertRow`? Would reduce the repetition for each row.

Also are we testing the change link HREF?

File: src/ui/server/helpers/summary-lists/your-business/index.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ * Create all fields for the insurance - Your business govukSummaryList
+ * @param {Object} All submitted your business data
+ * @returns {Object} All your business values in an object structure for GOVUK summary list structure
+ */
+const generateFields = (answers: ApplicationExporterCompany, referenceNumber: number) => {
+  let fields = [] as Array<SummaryListItemData>;
+
+  fields = generateYourCompanyFields(answers, referenceNumber);
+
+  fields = [...fields];
```
Comment (ttbarnes) [inline src/ui/server/helpers/summary-lists/your-business/index.ts:16]: No need for this line

File: src/ui/server/helpers/summary-lists/your-business/index.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+  return fields;
+};
+
+/**
+ * yourBusinessSummaryList
+ * Create multiple groups with govukSummaryList data structure
+ * @param {Object} All answers/submitted data in a simple object.text structure
```
Comment (ttbarnes) [inline src/ui/server/helpers/summary-lists/your-business/index.ts:24]: Missing param for reference number

File: src/ui/server/helpers/summary-lists/your-business/sic-code-mapping/index.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import { DEFAULT } from '../../../../content-strings';
+import { ApplicationExporterSicCodes } from '../../../../../types';
+
+/**
+ * sicCodeMapping
+ * maps application sic code response to generate a string of the sic code values
+ * @param {Array<ApplicationExporterSicCodes>} applicationSicCodes
+ * @returns {String} sicCodes
+ */
+const sicCodeMapping = (applicationSicCodes?: Array<ApplicationExporterSicCodes>) => {
```
Comment (ttbarnes) [inline src/ui/server/helpers/summary-lists/your-business/sic-code-mapping/index.ts:11]: How about `mapSicCodes`? Same with the file name

File: src/ui/server/helpers/summary-lists/your-business/sic-code-mapping/index.test.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  });
+
+  it('should return a string with one sic codes if provided with one sic code', () => {
+    const sicCodes = [
+      {
+        id: 'cle5xb2op0986dcno2yiidukj',
+        sicCode: '64999',
+        __typename: 'ExporterCompanySicCode',
+      },
+    ];
```
Comment (ttbarnes) [inline src/ui/server/helpers/summary-lists/your-business/sic-code-mapping/index.test.ts:34]: Do we have test mocks we can use for these? Also `one sic codes` should be `one sic code` 😛

File: src/ui/server/helpers/summary-lists/your-business/your-company-fields/index.ts [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+const {
+  TURNOVER: {
+    [FINANCIAL_YEAR_END_DATE]: { DATE_FORMAT },
+  },
+} = FIELDS;
+
+/**
+ * generateYourCompanyFields
+ * Create all your company fields and values for the Insurance - Company details govukSummaryList
+ * @param {Object} answers exporter company data
```
Comment (ttbarnes) [inline src/ui/server/helpers/summary-lists/your-business/your-company-fields/index.ts:34]: Missing reference number param

File: src/ui/templates/insurance/your-business/check-your-answers.njk [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+  <div class="govuk-grid-row">
+    <div class="govuk-grid-column-full-from-desktop">
+      <span class="govuk-caption-l" data-cy="heading-caption">{{ CONTENT_STRINGS.HEADING_CAPTION }}</span>
+      <h1 class="govuk-heading-xl" data-cy="heading">{{ CONTENT_STRINGS.PAGE_TITLE }}</h1>
+    </div>
+  </div>
+
+  {{ govukSummaryList({
+    classes: 'govuk-!-text-align-left',
```
Comment (ttbarnes) [inline src/ui/templates/insurance/your-business/check-your-answers.njk:29]: Might not need this class for insurance summary lists

File: src/ui/server/helpers/summary-lists/your-business/sic-code-mapping/index.ts (hunk 4 above)
Comment (Zainzzkk) [reply src/ui/server/helpers/summary-lists/your-business/sic-code-mapping/index.ts:11]: we already have a mapSicCodes for adding to the db

File: e2e-tests/cypress/fixtures/application.js [hunk 8]
```diff
@@ -75,8 +83,23 @@ const application = {
...
+    [COMPANY_NUMBER]: COMPANIES_HOUSE_NUMBER,
+    [COMPANY_NAME]: 'DHG PROPERTY FINANCE LIMITED',
+    [COMPANY_INCORPORATED]: '10 April 2014',
+    [COMPANY_ADDRESS]: {
+      addressLine1: 'Unit 3 Lewis Court',
+      addressLine2: 'Portmanmoor Road',
+      careOf: '',
+      locality: 'Cardiff',
+      region: 'South Glamorgan',
+      postalCode: 'CF24 5HQ',
```
Comment (ttbarnes) [inline e2e-tests/cypress/fixtures/application.js:95]: Can we use UKEF address?

File: src/ui/server/helpers/summary-lists/your-business/map-sic-codes/index.ts [hunk 9]
```diff
@@ -0,0 +1,21 @@
+import { generateSicCodesValue } from '../../company-house-summary-list';
+import { DEFAULT } from '../../../../content-strings';
+import { ApplicationExporterSicCodes } from '../../../../../types';
+
+/**
+ * sicCodeMapping
```
Comment (ttbarnes) [inline src/ui/server/helpers/summary-lists/your-business/map-sic-codes/index.ts:6]: typo

File: e2e-tests/cypress/fixtures/application.js (hunk 8 above)
Comment (Zainzzkk) [reply e2e-tests/cypress/fixtures/application.js:95]: address for the company, has to be exact

File: e2e-tests/cypress/e2e/journeys/insurance/your-business/change-your-answers/check-your-answers-change-company-details.spec.js [hunk 10]
```diff
@@ -0,0 +1,223 @@
...
+  },
+  START,
+} = ROUTES.INSURANCE;
+
+const { taskList } = partials.insurancePartials;
+
+const task = taskList.prepareApplication.tasks.exporterBusiness;
+
+const { summaryList } = checkYourAnswers;
+
```
Comment (ttbarnes) [inline e2e-tests/cypress/e2e/journeys/insurance/your-business/change-your-answers/check-your-answers-change-company-details.spec.js:42]: Can we change the file name to be `change-your-answers`

File: src/ui/server/controllers/insurance/business/company-details/index.POST-continue.test.ts [hunk 11]
```diff
@@ -71,26 +71,48 @@ describe('controllers/insurance/business/companies-details', () => {
...
 
     describe('when there are no validation errors', () => {
-      it('should redirect to next page', async () => {
-        req.body = {
-          [INPUT]: '8989898',
-          [TRADING_NAME]: 'true',
-          [TRADING_ADDRESS]: 'false',
-          [PHONE_NUMBER]: VALID_PHONE_NUMBERS.LANDLINE,
-        };
+      describe('when the originalUrl contains change', () => {
```
Comment (ttbarnes) [inline src/ui/server/controllers/insurance/business/company-details/index.POST-continue.test.ts:82]: Can we align these describes to be aligned with other answers tests 

```js
// default 
it(`should redirect to ${ABOUT_GOODS_OR_SERVICES}`, async () => {


describe("when the url's last substring is `change`", () => {
  it(`should redirect to ${CHECK_YOUR_ANSWERS}`, async () => {


```

File: src/ui/server/helpers/summary-lists/your-business/index.ts [hunk 12]
```diff
@@ -0,0 +1,35 @@
+import generateSummaryListRows from '../generate-summary-list-rows';
+import generateYourCompanyFields from './your-company-fields';
+import { ApplicationExporterCompany, SummaryListItemData } from '../../../../types';
+
+/**
+ * generateFields
+ * Create all fields for the insurance - Your business govukSummaryList
+ * @param {ApplicationExporterCompany} answersExporterCompany application exporterCompany object
+ * @param {number} referenceNumber application reference number
```
Comment (ttbarnes) [inline src/ui/server/helpers/summary-lists/your-business/index.ts:9]: `Number` same below

File: src/ui/server/helpers/summary-lists/your-business/map-sic-codes/index.ts [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+import { ApplicationExporterSicCodes } from '../../../../../types';
+
+/**
+ * mapSicCodes
+ * maps application sic code response to generate a string of the sic code values
+ * @param {Array<ApplicationExporterSicCodes>} applicationSicCodes
+ * @returns {String} sicCodes
+ */
+const mapSicCodes = (applicationSicCodes?: Array<ApplicationExporterSicCodes>) => {
+  if (!applicationSicCodes || !applicationSicCodes.length) {
```
Comment (ttbarnes) [inline src/ui/server/helpers/summary-lists/your-business/map-sic-codes/index.ts:12]: Can use `isPopulatedArray` helper here - then only need one unit tests for if array is not populated / is empty array

File: src/ui/server/helpers/summary-lists/your-business/index.ts (hunk 12 above)
Comment (Zainzzkk) [reply src/ui/server/helpers/summary-lists/your-business/index.ts:9]: do you mean ` referenceNumber: number` as below?

File: src/ui/server/helpers/summary-lists/your-business/map-sic-codes/index.ts (hunk 13 above)
Comment (Zainzzkk) [reply src/ui/server/helpers/summary-lists/your-business/map-sic-codes/index.ts:12]: this causes eslint to have fun

File: src/ui/server/helpers/summary-lists/your-business/your-company-fields/index.ts [hunk 14]
```diff
@@ -0,0 +1,112 @@
...
+  TURNOVER: {
+    [FINANCIAL_YEAR_END_DATE]: { DATE_FORMAT },
+  },
+} = FIELDS;
+
+/**
+ * generateYourCompanyFields
+ * Create all your company fields and values for the Insurance - Company details govukSummaryList
+ * @param {ApplicationExporterCompany} answers exporter company data
+ * @param {number} referenceNumber application reference number
```
Comment (ttbarnes) [inline src/ui/server/helpers/summary-lists/your-business/your-company-fields/index.ts:35]: typo

File: src/ui/server/controllers/insurance/business/company-details/index.test.ts [hunk 15]
```diff
@@ -49,20 +55,44 @@ describe('controllers/insurance/business/companies-details', () => {
...
-          COMPANIES_HOUSE: `${INSURANCE_ROOT}/${mockApplication.referenceNumber}${COMPANY_HOUSE_SEARCH}`,
-          COMPANY_DETAILS: `${INSURANCE_ROOT}/${mockApplication.referenceNumber}${COMPANY_DETAILS_ROUTE}`,
-          NO_COMPANIES_HOUSE_NUMBER: `${INSURANCE_ROOT}/${mockApplication.referenceNumber}${NO_COMPANIES_HOUSE_NUMBER}`,
-          SAVE_AND_BACK_URL: `${INSURANCE_ROOT}/${mockApplication.referenceNumber}${COMPANY_DETAILS_SAVE_AND_BACK}`,
-        },
-        FIELDS: EXPORTER_BUSINESS,
-      };
-
-      expect(result).toEqual(expected);
+    describe('originalUrl does not contain "change"', () => {
```
Comment (ttbarnes) [inline src/ui/server/controllers/insurance/business/company-details/index.test.ts:58]: Not a blocker for this PR, but can we align with the other descriptions, it's not "contain" 😄

File: src/ui/server/helpers/summary-lists/your-business/index.ts [hunk 16]
```diff
@@ -0,0 +1,35 @@
+import generateSummaryListRows from '../generate-summary-list-rows';
+import generateYourCompanyFields from './your-company-fields';
+import { ApplicationExporterCompany, SummaryListItemData } from '../../../../types';
+
+/**
+ * generateFields
+ * Create all fields for the insurance - Your business govukSummaryList
+ * @param {ApplicationExporterCompany} answersExporterCompany application exporterCompany object
+ * @param {Number} referenceNumber application reference number
```
Comment (ttbarnes) [inline src/ui/server/helpers/summary-lists/your-business/index.ts:9]: `Application reference number`?

**Review by ttbarnes — APPROVED** [review]
